### PR TITLE
Add device registry to MQTT binary sensor

### PR DESCRIPTION
**Pull request by OttoWinter**
**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#17243

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/binary_sensor.mqtt.markdown [hunk 1]
```diff
@@ -115,6 +115,35 @@ force_update:
...
+  description: 'Information about the device this binary sensor is a part of to tie it into the [device registry](https://developers.home-assistant.io/docs/en/device_registry_index.html). Only works through [MQTT discovery](/docs/mqtt/discovery/) and when [`unique_id`](#unique_id) is set.'
+  required: false
+  type: map
+  keys:
+    identifiers:
+      description: 'A list of IDs that uniquely identify the device. For example a serial number.'
+      required: false
+      type: list, string
+    connections:
+      description: 'A list of connections of the device to the outside world as a list of tuples `[connection_type, connection_identifier]`. For example the MAC address of a network interface: `[["mac", "02:5b:26:a8:dc:12"]]`.'
```
Comment (DubhAd) [inline source/_components/binary_sensor.mqtt.markdown]: Should that be single, or double, brackets?

**Review by DubhAd — COMMENTED** [review]
Connections example is confusing, showing both `[ blah, blah]` an `[[blah, blah]]`

File: source/_components/binary_sensor.mqtt.markdown (hunk 1 above)
Comment (OttoWinter) [reply source/_components/binary_sensor.mqtt.markdown]: double brackets. It's a list of 2-tuples.

Comment (OttoWinter) [reply source/_components/binary_sensor.mqtt.markdown]: I clarified it a bit now by including `"connections": ` in the example config. I'm writing it in JSON format as that's the only way this config option can be used (only over MQTT discovery).

**Review by DubhAd — APPROVED** [review]
👍

**Pull request merged**